utils/remote.cc

```diff
@@ -2280,117 +2228,118 @@ void filter_ids(tr_variant::Map const& map, RemoteConfig& config)
     }
 }
 
-int process_response(char const* rpcurl, std::string_view response, RemoteConfig& config)
+int process_response(char const* rpcurl, std::string_view const response, RemoteConfig& config)
 {
-    auto status = int{ EXIT_SUCCESS };
+    if (config.json)
+    {
+        fmt::print("{:s}\n", response);
+        return EXIT_SUCCESS;
+    }
 
     if (config.debug)
     {
         fmt::print(stderr, "got response (len {:d}):\n--------\n{:s}\n--------\n", std::size(response), response);
     }
 
-    if (config.json)
+    auto parsed = tr_variant_serde::json().inplace().parse(response);
+    if (!parsed)
     {
-        fmt::print("{:s}\n", response);
-        return status;
+        fmt::print(stderr, "Unable to parse response '{}'\n", response);
+        return EXIT_FAILURE;
     }
+    api_compat::convert_incoming_data(*parsed);
 
-    if (auto top = tr_variant_serde::json().inplace().parse(response); !top)
+    auto const* const top = parsed->get_if<tr_variant::Map>();
+    if (!top)
     {
-        fmt::print(stderr, "Unable to parse response '{}'\n", response);
-        status |= EXIT_FAILURE;
+        fmt::print(stderr, "Response was not a JSON object\n");
+        return EXIT_FAILURE;
     }
-    else if (auto* map_ptr = top->get_if<tr_variant::Map>(); map_ptr == nullptr)
+
+    if (!top->contains(TR_KEY_jsonrpc))
```

Comment:
```suggestion
    if (auto const jsonrpc = top->value_if<std::string_view>(TR_KEY_jsonrpc); jsonrpc != JsonRpc::Version)
```

Reply:
6ec1dc089b5c47301fdd13b78d1fdb7a500295be

Reply:
Ah, nice that you reference `JsonRpc::Version` instead of hard coding the version into the log string. 👍🏻